### DISCUSSION TITLE
Organisation API for n8n

### DISCUSSION THREAD

**Discussion by backendoadmin** [dani-garcia/vaultwarden#2317] [Q&A] [answered]
Hello!

thank you for maintaining this great app. I started using n8n as well, and I'd like to connect to vaultwarden. And I see there is cli api, but there is no api key for the organization. Did I miss something? Is there a way to add this feature?

https://docs.n8n.io/credentials/bitwarden/#using-api-key

I think it would be very helpful. Just let me know your toughts. Thank you

**Comment by jjlin** [comment, marked as answer]
As noted at https://github.com/dani-garcia/vaultwarden/wiki#missing-features, support for the Bitwarden Public API has not been implemented. I've updated the wiki to mention Organization API keys not being implemented as well, as those are only used with the public API.

These features could certainly be added, just no one has found the time and/or motivation to do it yet.

**Comment by mwatermolen** [comment]
Hate to bump an old issue, but I found the motivation https://github.com/n8n-io/n8n/pull/3780

**Reply by BlackDex** [reply]
That's not really a motivation i think.
Though, that said, I have made a start on the org apikey already.
Mostly to allow the usage of the latest version of the Directory connector.

See https://github.com/BlackDex/vaultwarden/tree/org-api-key

**Reply by mwatermolen** [reply]
\*\* I found the motivation to fix n8n

**Reply by BlackDex** [reply]
Ah!

**Reply by mwatermolen** [reply]
Out of curiosity, how for off are you from a PR

**Reply by BlackDex** [reply]
Well, since the organizational APIKey is only useful for the public-api, and that is not implemented at all, it kinda takes a while i think.
As said, I plan to only allow the usage of the latest Directory Connector for now, but that probably will make a start for the public-api endpoints.

So, i have no specific time frame.

**Reply by mwatermolen** [reply]
Thanks for the explanation! Keep up the great work!

**Reply by rizlas** [reply]
> Well, since the organizational APIKey is only useful for the public-api, and that is not implemented at all, it kinda takes a while i think.
> As said, I plan to only allow the usage of the latest Directory Connector for now, but that probably will make a start for the public-api endpoints.

If I understand correctly using the CLI with organization api key will not work cause some Public API are not implemented right? Example of error with `bw login --apikey`: `Invalid API Key; Organization API Key currently not supported`

**Reply by BlackDex** [reply]
That is correct. It is only used for the directory connector right now.

**Comment by backendoadmin** [comment]
I remain highly interested in this topic, and to my delight, I discovered that there was a relevant commit made just yesterday. 
https://github.com/dani-garcia/vaultwarden/commit/05cf343704e30a449b2cdc3560130ae128eb51cd 
Keep up the good work! :)
